homeassistant/components/webostv/__init__.py

```diff
@@ -98,55 +141,43 @@ async def async_service_handler(service):
             DOMAIN, service, async_service_handler, schema=schema
         )
 
-    tasks = [async_setup_tv(hass, config, conf) for conf in config[DOMAIN]]
-    if tasks:
-        await asyncio.gather(*tasks)
-
-    return True
-
-
-def convert_client_keys(config_file):
-    """In case the config file contains JSON, convert it to a Sqlite config file."""
-    # Return early if config file is non-existing
-    if not os.path.isfile(config_file):
-        return
+    hass.data[DOMAIN][config_entry.entry_id] = client
+    hass.config_entries.async_setup_platforms(config_entry, PLATFORMS)
 
-    # Try to parse the file as being JSON
-    with open(config_file) as json_file:
-        try:
-            json_conf = json.load(json_file)
-        except (json.JSONDecodeError, UnicodeDecodeError):
-            json_conf = None
-
-    # If the file contains JSON, convert it to an Sqlite DB
-    if json_conf:
-        _LOGGER.warning("LG webOS TV client-key file is being migrated to Sqlite!")
-
-        # Clean the JSON file
-        os.remove(config_file)
-
-        # Write the data to the Sqlite DB
-        with SqliteDict(config_file) as conf:
-            for host, key in json_conf.items():
-                conf[host] = key
-            conf.commit()
+    # set up notify platform, no entry support for notify component yet,
+    # have to use discovery to load platform.
+    hass.async_create_task(
+        discovery.async_load_platform(
+            hass,
+            "notify",
+            DOMAIN,
+            {
+                CONF_NAME: config_entry.data[CONF_NAME],
+                ATTR_CONFIG_ENTRY_ID: config_entry.entry_id,
+            },
+            hass.data[DOMAIN],
+        )
+    )
 
+    if not config_entry.update_listeners:
+        config_entry.async_on_unload(
+            config_entry.add_update_listener(async_update_options)
+        )
 
-async def async_setup_tv(hass, config, conf):
-    """Set up a LG WebOS TV based on host parameter."""
+    async def async_on_stop(_event):
+        """Unregister callbacks and disconnect."""
+        client.clear_state_update_callbacks()
+        await client.disconnect()
 
-    host = conf[CONF_HOST]
-    config_file = hass.config.path(WEBOSTV_CONFIG_FILE)
-    await hass.async_add_executor_job(convert_client_keys, config_file)
```

Comment:
Do we expect they are all converted at this point?

Will this be a breaking change for users who haven't upgraded recently?

Reply:
Migration of keys was introduced at 2021.4.0, in my opinion it does not worth keeping this migration code, but listing in breaking changes that if updating from version before 2021.4.0 the TV will need to be re-paired.